### PR TITLE
lint: Add linting to the build

### PR DESCRIPTION
**Pull request by aeisenberg**
This PR ensures all files are linted properly and (I think) adds linting to the build process. The build should fail if there are linting errors.

### CODE REVIEW

File: extensions/ql-vscode/src/databases.ts [hunk 1]
```diff
@@ -430,21 +430,23 @@ class DatabaseItemImpl implements DatabaseItem {
...
       logger.log(`Waiting for event ${event} timed out after ${timeoutMs}ms`);
       res(undefined);
       dispose();
     }, timeoutMs);
+    const disposable = event(e => {
+      res(e);
+      dispose();
+    });
+    function dispose() {
+      if (timeout !== undefined) {
```
Comment (jcreedcmu) [inline extensions/ql-vscode/src/databases.ts]: Huh, somehow I didn't think typescript let me forward-reference this function recursively. If it does, these checks for `!== undefined` aren't necessary anymore, are they?

File: extensions/ql-vscode/src/quick-query.ts [hunk 2]
```diff
@@ -70,6 +70,16 @@ function getQuickQueriesDir(ctx: ExtensionContext): string {
   return queriesPath;
 }
 
+
+function updateQuickQueryDir(queriesDir: string, index: number, len: number) {
```
Comment (jcreedcmu) [inline extensions/ql-vscode/src/quick-query.ts]: Why is this required to be hoisted? I'm not really against it, just curious.

Comment (aeisenberg) [reply extensions/ql-vscode/src/quick-query.ts]: There was a linting warning to avoid nested functions. It's not really a big deal and I don't feel strongly about it. It was easier for me to change the code than to change the rule.

File: extensions/ql-vscode/src/databases.ts (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/databases.ts]: `timeout !== undefined` is never needed since `clearTimeout` is a noop for non-timers. And it looks like the other if statement is not necessary either.

File: extensions/ql-vscode/src/quick-query.ts (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/quick-query.ts]: Actually, now that I look at the rule more closely, it specifically targets named functions that are not in the top-level of another function or global. I think this is generally good since it makes things harder to read if there are lots of nested functions scattered inside of if statements and elsewhere. Made a little fix here.

**Comment by jcreedcmu** [comment]
Curious why it didn't complain about the nested `function dispose()` then. Not terrifically important, although I'd be in favor of allowing nested functions generally.

**Comment by aeisenberg** [comment]
Functions that are top level inside another one are ok. https://eslint.org/docs/rules/no-inner-declarations

```
function a() {
  function b()  {}
}
```
is ok, but

```
function a() {
  try {
    function b()  {}
  } catch (e) {}
}
```
is not.
